Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1

```diff
@@ -618,4 +618,53 @@ function DemistoVersionGreaterEqualThen([string]$version) {
     $version = (Select-string -Pattern $version_pattern -InputObject $version).Matches[0].Value
 
     return [version]::Parse($demisto_version) -ge  [version]::Parse($version)
-}
\ No newline at end of file
+}
+
+function ParseDateRange{
+    [CmdletBinding()]
+    [OutputType([System.Object[]])]
+    Param (
+        [parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $date_str
+    )
+    $now = $date = Get-Date
+    $number, $unit_name = $date_str.Split()
+    try{
+        $number = -([int]$number)
+    } catch [System.Management.Automation.RuntimeException]{
+        throw "No number given in $date_str"
+    }
+    if ($null -eq $unit_name){
+        throw "Time unit not given in $date_str"
+    }
+    if (!($unit_name.GetType() -eq [String])) {
+        throw "Too many arguemnts in $date_str"
+    }
+    if ($unit_name.Contains("minute")){
+        $date = $date.AddMinutes($number)
+    } elseif ($unit_name.Contains("hour")) {
+        $date = $date.AddHours($number)
+    } elseif ($unit_name.Contains("day")){
+        $date = $date.AddDays($number)
+    } elseif ($unit_name.Contains("week")) {
+        $date = $date.AddDays($number * 7)
+    } elseif ($unit_name.Contains("month")) {
+        $date = $date.AddMonths($number)
+    } elseif ($unit_name.Contains("year")) {
+        $date = $date.AddYears($number)
+    } else {
+        throw "Could not process time unit $unit_name"
```

Comment:
Validate that in the beginning and specify which unit types are excepted i.e (minute\s, hour\s, day\s etc.)